Assignment4/MyLibrary/MyLibrary.Tests/NumSetTests.cs

```diff
@@ -0,0 +1,239 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+
+namespace MyLibrary.Tests
+{
+    [TestClass]
+    public class NumSetTests
+    {
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void NumSet_PassesNullArgumentsToConstructor_ThrowsArgumentNullException()
+        {
+            //Assign
+
+            //Act
+            _ = new NumSet(null!);
+
+            //Assert
+        }
+
+        [TestMethod]
+        public void NumSet_PassesArrayToConstructor_AssignsArrayToHashSet()
+        {
+            //Assign
+            int[] array = new int[] { 1, 2, 3, 4, 5 };
+            NumSet numSet = new NumSet(array);
+
+            //Act
+            _ = new HashSet<int>(array);
+
+            //Assert
+            Assert.IsNotNull(numSet.MySet);
+        }
+
+        [TestMethod]
+        public void ToString_TakesHashSet_ReturnsStringOfHashSet()
+        {
+            //Assign
+            int[] array = new int[] { 1, 2, 3, 4, 5 };
+            NumSet numSet = new NumSet(array);
+
+            //Act
+            string arrayToString = numSet.ToString();
+
+            //Assert
+            Assert.AreEqual(arrayToString, "1, 2, 3, 4, 5");
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void ToString_TakesNullHashSet_ThrowsArgumentNullException()
+        {
+            //Assign
+            NumSet? numSet = new NumSet
+            {
+                MySet = null
+            };
+
+            //Act
+            _ = numSet.ToString();
+
+            //Assert
+        }
+
+        [TestMethod]
+        public void Equals_PassesDifferentNumSets_ReturnsFalse()
+        {
+            //Assign
+            int[] array1 = new int[] { 1, 2, 3, 4, 5 };
+            int[] array2 = new int[] { 6, 7, 8, 9, 10 };
+            NumSet numSet1 = new NumSet(array1);
+            NumSet numSet2 = new NumSet(array2);
+
+            //Act
+
+            //Assert
+            Assert.IsFalse(numSet1.Equals(numSet2));
+        }
+
+        [TestMethod]
+        public void Equals_PassesTheSameNumSet_ReturnsTrue()
+        {
+            //Assign
+            int[] array = new int[] { 1, 2, 3, 4, 5 };
+            NumSet numSet = new NumSet(array);
+
+            //Act
+
+            //Assert
+            Assert.IsTrue(numSet.Equals(numSet));
```

Comment:
```suggestion
            //Assign
            int[] array = new int[] { 1, 2, 3, 4, 5 };
            NumSet numSet1 = new NumSet(array);
            NumSet numSet2 = new NumSet(array);

            //Act

            //Assert
            Assert.IsTrue(numSet1.Equals(numSet2)); // This should not fail
```
This will fail because of the `==` being used to compare in your override of `Equals`.